File: Test/m/cstr.m
function xdot = dynamics(t, x, u)
    q = 1; V = 1; CA0 = 1; T0 = 350; Tc0 = 300;
    k0 = 7.2e10; E_R = 8750; dH = -2e5; rho = 1000; Cp = 0.239;
    UA = 5e4; rhoc = 1000; Cpc = 1; Vc = 1;
    CA = x(1); T = x(2); Tc = x(3);
    r = k0 * exp(-E_R/T) * CA;
    xdot = [q/V*(CA0 - CA) - r;
            q/V*(T0 - T) - dH/(rho*Cp)*r + UA/(V*rho*Cp)*(Tc - T);
            u/Vc*(Tc0 - Tc) + UA/(Vc*rhoc*Cpc)*(T - Tc)];
end

File: Test/py/cstr.py
from numpy import *

def dynamics(t, x, u):
    q, V, CA0, T0, Tc0 = 1, 1, 1, 350, 300
    k0, E_R, dH, rho, Cp = 7.2e10, 8750, -2e5, 1000, 0.239
    UA, rhoc, Cpc, Vc = 5e4, 1000, 1, 1
    CA, T, Tc = x[0], x[1], x[2]
    r = k0 * exp(-E_R/T) * CA
    xdot = array([[q/V*(CA0 - CA) - r],
            [q/V*(T0 - T) - dH/(rho*Cp)*r + UA/(V*rho*Cp)*(Tc - T)],
            [u/Vc*(Tc0 - Tc) + UA/(Vc*rhoc*Cpc)*(T - Tc)]])
    return xdot